tracing-subscriber/src/registry/sharded.rs

```diff
@@ -270,3 +320,100 @@ impl<'a> SpanData<'a> for Data<'a> {
         ExtensionsMut::new(self.inner.extensions.write().expect("Mutex poisoned"))
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::Registry;
+    use crate::{layer::Context, registry::LookupSpan, Layer};
+    use std::sync::atomic::{AtomicBool, Ordering};
+    use tracing::{self, subscriber::with_default};
+    use tracing_core::{
+        span::{Attributes, Id},
+        Subscriber,
+    };
+
+    struct NopLayer;
```

Comment:
Nit: I wouldn't call this a "nop" layer; it makes an assertion...not a blocker.

Reply:
called it "AssertionLayer" and "AssertionLayer2".

Reply:
Actually, removed `AssertionLayer2` and composed it twice. Works well!